internal/kubernetes_test.go

```diff
@@ -349,22 +350,21 @@ func TestKubeInvalidConfig3(t *testing.T) {
 }
 
 func TestKubeInvalidSecretType(t *testing.T) {
-	testRequestKube(t, &Exporter{
-		KubeIncludeNamespaces: []string{"default"},
-		KubeSecretTypes:       []string{"aze"},
-	}, func(m []model.MetricFamily) {
-		metrics := getMetricsForName(m, "x509_read_errors")
-		assert.Equal(t, 1., metrics[0].GetGauge().GetValue())
-	})
+	_, err := ParseSecretType("aze")
+	assert.Error(t, err)
 }
 
 func TestKubeEmptyStringKey(t *testing.T) {
 	testRequestKube(t, &Exporter{
 		KubeIncludeLabels: []string{"empty=true"},
-		KubeSecretTypes:   []string{"kubernetes.io/tls:tls.crt", "kubernetes.io/tls:tls.key", "kubernetes.io/tls:nil.key"},
+		KubeSecretTypes: []KubeSecretType{
+			{Type: "kubernetes.io/tls", Regexp: regexp.MustCompile(`tls\.crt`)},
+			{Type: "kubernetes.io/tls", Regexp: regexp.MustCompile(`tls\.key`)},
+			{Type: "kubernetes.io/tls", Regexp: regexp.MustCompile(`nil\.key`)},
+		},
 	}, func(m []model.MetricFamily) {
 		metrics := getMetricsForName(m, "x509_read_errors")
-		assert.Equal(t, 0., metrics[0].GetGauge().GetValue())
+		assert.Equal(t, 1., metrics[0].GetGauge().GetValue())
```

Comment:
Why does the behavior of `getMetricsForName` change? It should stay the same

Reply:
The current implementation removed the length check ` len(secret.Data[typeAndKey[1]]) > 0 ` on secret values. So, zero length values are returned from `parseAllKubeSecrets()`, which in turn cause parse errors. I could add these checks again.